packages/ionio/src/Transaction.ts

```diff
@@ -280,17 +322,31 @@ export class Transaction implements TransactionInterface {
     }
 
     // check for blinding to be made
-    if (this.inputBlindingData.size > 0) {
-      if (this.outputBlindingPubKeys.size === 0)
+    if (this.inputBlinders.length > 0) {
+      const needsBlinding = this.pset.outputs.some((o: PsetOutput) =>
+        o.needsBlinding()
+      );
+      if (!needsBlinding)
```

Comment:
There should already be an api for this:
```suggestion
      if (!this.pset.needsBlinding())
```